System/Posix/Files/Common.hsc

```diff
@@ -286,6 +291,22 @@ modificationTime (FileStatus stat) =
 statusChangeTime (FileStatus stat) =
   unsafePerformIO $ withForeignPtr stat $ (#peek struct stat, st_ctime)
 
+#ifdef HAVE_STRUCT_STAT_ST_BLOCKS
+fileBlocks (FileStatus stat) =
+  Just $ unsafePerformIO $ withForeignPtr stat $ (#peek struct stat, st_blocks)
+#else
+# WARNING fileBlocks "fileBlocks: not available on this platform" #
```

Comment:
What is the purpose of the WARNING? It will be seen by anyone compiling the package on such a platform, whether they intend to use the function or not, and will not be seen by the caller, where the `Nothing` result adequately conveys the same message in a more targeted manner.  So unless I'm missing something the WARNING is not useful and should be removed.

Reply:
Hmm, I think I originally had these as `{-# WARNING #-}` pragmas, which would fire if the function is used, but I agree it doesn't make much sense as a CPP warning.